### PR TITLE
build: Rework CI and move all bundling into new build target.

### PR DESCRIPTION
**Pull request by Steveice10**
Large scale rework of the CI layout, scripts, and bundling.

Breaking it down point-by-point:
* Introduces CMake bundling support for the remaining platforms. Establishes a convention where bundling will put outputs in the "bundle" folder within the build directory, including all of the necessary binaries along with the license, readme, and extra scripts.
  * Windows outputs all of the executables with their required DLLs.
  * macOS creates an app bundle for citra-qt and a folder of libraries for the standalone SDL citra.
  * Linux creates an AppImage for each target executable.
* Introduces a new "bundle" target for all of the app bundling logic, rather than doing it in the main build path. This keeps local development builds nice and quick while also allowing for local builders to create full bundles for testing or sharing. It also means if we change CIs in the future we can carry forward a lot of this logic as part of the CMake build process.
* Restructures the CI to include the following builds:
  * Android: Builds and outputs the APK and AAB files for Citra Android.
    * Note that release signing support was moved to use the built-in Gradle signing configurations instead of requiring us to manually sign after the build completes. There's also a new Gradle task to create and copy files to the "bundle" output directory we use for other build types to keep it consistent.
  * Windows:
    * MSVC: Builds and outputs a distributable bundle using the MSVC toolchain.
    * MSYS2: Builds and outputs a distributable bundle using the MSYS2 Clang64 toolchain and libraries.
  * Linux
    * AppImage: Builds and outputs a distributable bundle for Linux users using AppImages.
    * Fresh: Verifies build against the latest fresh Docker image, to make sure users can build against newer tools and libraries. Does not upload any artifacts.
  * macOS: Builds for x86_64 and ARM64, then combines into a universal distributable, same as current.
* Other clean-ups and enhancements to the CI, trimming down the scripts and bringing them up to date.
  * For example, we now use CMAKE\_\<LANG\>\_COMPILER\_LAUNCHER for ccache instead of prefixing CMAKE\_\<LANG\>\_COMPILER or setting environment variables.
  * System/Bundled library CMake flags have been standardized and grouped together for clarity.
  * FFmpeg AAC dynamic loading support has been removed. The video dumper directly links without dynamic loading, which makes this useless, and we provide the FFmpeg DLLs with the release bundles.

This will conflict with the Vulkan PR so we can't canary it yet, and it will need to be rebased to include a few of the updates there like MoltenVK support. But I'm opening this now so I can start getting review and some testing against the PR artifacts.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6556)

### CODE REVIEW

**Comment by Steveice10** [comment]
(note that this will need to be checked against the Qt installer given the artifact output changes)

**Comment by fight-fly** [comment]
Due to ffmpeg, the total size of these packaged files is too large. Some files are not used, such as rav1e.dll.

Can the size be reduced.
Thank you.

BTW, my system is Windows.

**Comment by Steveice10** [comment]
> Due to ffmpeg, the total size of these packaged files is too large. Some files are not used, such as rav1e.dll.
> 
> Can the size be reduced. Thank you.
> 
> BTW, my system is Windows.

These are part of the ffmpeg build provided by MSYS2 and are used as codecs for the video dumper feature. We'd likely need to move to our own FFmpeg builds for the MSYS2 builds to cut them down.

File: src/web_service/verify_login.cpp [hunk 1]
```diff
@@ -22,7 +22,7 @@ bool VerifyLogin(const std::string& host, const std::string& username, const std
         return username.empty();
     }
 
-    return username == *iter;
+    return username == (*iter).get<std::string>();
```
Comment (SachinVin) [inline src/web_service/verify_login.cpp:25]: `iter->get<std::string>();` ?

**Review by SachinVin — APPROVED** [review]

**Comment by GPUCode** [comment]
Is this taggable after the vulkan prepare PR?

**Comment by Steveice10** [comment]
> Is this taggable after the vulkan prepare PR?

I will need to check after I rebase next, we might need to work something out around the change to re-enable macOS CI in Vulkan.

**Comment by Steveice10** [comment]
Rebased and added a compatibility workaround for the MSYS2 release names to keep current Qt installer setups working smoothly.

**Comment by Steveice10** [comment]
Untagging due to conflict, will re-tag after #6618 is in.

**Comment by vitor-k** [comment]
Does the installer work with the naming change from osx to macos?

**Comment by Steveice10** [comment]
> Does the installer work with the naming change from osx to macos?

Good catch. I think closer to merge we can go and update the build names in the repo generating code, for now I’ll add a compatibility bit similar to mingw.

**Comment by SachinVin** [comment]
Can't we just update the installer to pick the new names too, we can probably just merger that change to nightly

**Pull request merged**